### ISSUE TITLE
BUG - Non-administrative users are unable to view recipes

### ISSUE THREAD

**Issue by djmega152** [mealie-recipes/mealie#2685]
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] This issue can be replicated on the demo site (https://demo.mealie.io/)

### What is the issue you are experiencing?

When a non-adminstrative user is added to a custom group, they are unable to view any recipes in the database.  I tried adding a new recipe after the group & user were added and the user was still not able to see the recipe

### Steps to Reproduce

1. Create a new group and enable all of the options.
![image](https://github.com/mealie-recipes/mealie/assets/99622018/1d72128d-4d81-4c83-ab26-ea6ea68367ce)
2. Add a user and assign them to the group.  I have tried with and without advanced features enabled.  All other options are not selected.
![image](https://github.com/mealie-recipes/mealie/assets/99622018/1369bee8-c710-4b05-a592-5650b6f6254d)
3. Logout as the admin and login as the test user.  Notice that no recipes are accessible. 

### Please provide relevent logs

**Details**
Version: v1.0.0beta-5
Build: ee121a12f8db33ecb4db5f8582f7ea9788d019e4
Application Mode: Production
Demo Status: Not Demo
API Port: 9000
API Docs: Enabled
Database Type: sqlite
Recipe Scraper Version: 14.20.0

**Checks**
Secure Site: Yes
Server Side Base URL: Yes
LDAP Ready: No
Email Configured: Yes
Docker Volumes: success

### Deployment

Docker (Synology)

**Comment by michael-genson** [comment]
Are you able to reproduce this on the new RC1 build or the demo site? I remember this was an issue fixed a while ago on an older version. Beta 5 is about a year old now

**Comment by djmega152** [comment]
I did reproduce the issue on the demo site.  You'll see a new mnacation
group with a user of tryit.

**Comment by michael-genson** [comment]
It doesn't look like there are any recipes on the new group. Recipes aren't shared between groups, they're isolated instances, recipes can only be shared within a group

**Comment by djmega152** [comment]
So a dumb question (probably), how do you set a recipe as not private?  I
followed this chart but I don't see anything in the docs (probably missing
it) about how to set this on the recipe itself?

https://nightly.mealie.io/documentation/getting-started/faq/#how-do-private-groups-and-recipes-work

And one last question, is it possible to have read only users that can't
create recipes?  I couldn't find documentation on this?

**Comment by michael-genson** [comment]
For a recipe to be viewable without logging in, two criteria have to be met:
1. The group has to be non-private (https://demo.mealie.io/group/)
2. The recipe has to be public (see below, while in edit mode on a recipe): 
![image](https://github.com/mealie-recipes/mealie/assets/71845777/b3d953b4-d6cf-4291-8839-cf4f49a51bdf)

As an additional tool, you can set the default behavior for new recipes (so if you want recipes to be public/private, you can change the default behavior). This is in the same place as the group settings above. Note that it's only for _new_ recipes, not existing ones.

> is it possible to have read only users that can't create recipes?
I don't think there's a way to restrict users from managing recipes, but if you're on the RC build (not beta 5) you can browse public recipes without logging in: https://demo.mealie.io/explore/recipes/home (this link can be found on your instance by clicking on your name and clicking "Get Public Link". This only works if your group is not private)

**Comment by djmega152** [comment]
Interesting, my group is not private and all of my recipes are public.
I'll upgrade and see if the problem is resolved by the new version.  I used
Marius Hosting's instructions for installing and his stacks usually use
:latest for the image but in this case, it was hard-coded.

I'll backup the DB before upgrading just in case.

I'll also log an enhancement request to allow read-only users.  I want to
share the DB but not necessarily allow others to add to it.

Appreciate the quick responses here.  I love the package and am hoping for
an android app some day!

**Comment by michael-genson** [comment]
If you're on beta 5 the public recipe browser isn't available, so upgrading will fix that. However if you're logged in, you're only going to see that group's recipes, you won't see other groups' recipes (even if they're public)

**Comment by boc-the-git** [comment]
@djmega152 once you upgrade, please give the PWA on Android a go. i.e. when looking at the site in your browser, "install" it (you'll probably get a popup suggestion, or check the menu)

I've been using it for a little while now and it's working well! (more eyes on it wouldn't hurt of course)

**Comment by djmega152** [comment]
Can I ask a dumb question?  Is the release noted in this doc available on
Synology docker?

https://nightly.mealie.io/documentation/getting-started/installation/installation-checklist/

I'm very much a noob and so installing without a guide like Marius provides
is way above my skill level once I hit an error.  I tried exporting the
existing container settings, modifying the json, deleting the existing
container and then recreating by importing the modified json.  The
container created when I used hkotel/mealie:latest as the image but the API
container would not start (404 error) and the frontend one I didn't even
attempt to start without the API container.

I modified the container to try to pull the image from
ghcr.io/mealie-recipes/mealie:nightly but I got an error saying that image
does not exist.

**Comment by michael-genson** [comment]
Not a dumb question at all! Are you using the new container manager, or are you still on the old Docker app? (And, if you're not using container manager, are you able to upgrade to DSM 7.2?)

---

If you're on container manager, you can use docker compose natively (they're called Projects):
![image](https://github.com/mealie-recipes/mealie/assets/71845777/cc5b4a8c-efaf-45e3-b253-7d64682505b0)

![image](https://github.com/mealie-recipes/mealie/assets/71845777/4e3f8a1e-4969-46fc-8691-ffa8270d8e45)

On Synology I had to tweak the docker compose slightly - I hardcoded the mounted volume:
```yaml
...
volumes:
      - ./data:/app/data/
...
```

With this change you'll need to create a "data" folder wherever you decide to put your project (I chose to put it in a docker folder, but you can put it wherever you'd like).

So I set my project path to `/docker/mealie` and I created a `data` folder in `mealie`.

---

If you're using the old Docker app (because you can't get to DSM 7.2) you have to access your Synology via SSH and do it the old fashioned way through shell

**Comment by djmega152** [comment]
I have 7.2 and am using the new container manager but I actually found a
new install guide for v1 on Marius' site so I deployed that.  So onto a new
problem!

I'm trying to follow the migration guide
<https://nightly.mealie.io/documentation/getting-started/migrating-to-mealie-v1/>
but my export from the beta version doesn't match the pre v1 layout so
nothing happens when I click submit.  How do I get all of my old recipes
back?

This is the format of the export I did from the old instance:
[image: image.png]

**Comment by boc-the-git** [comment]
Going from v1-beta to v1 (nightly/RC1) you don't need to do the migration, you'd just point your new containers at the same data.
This is the relevant guide: https://nightly.mealie.io/documentation/getting-started/installation/installation-checklist/#migrating-from-other-v1-versions

These are my before and after docker-compose files from when I did the upgrade.. using SQLite, where the volume is where the data was being stored.

Before:
```
version: "2"
services:
  mealie-frontend:
    image: hkotel/mealie:frontend-v1.0.0beta-5
    container_name: mealie-frontend
    environment:
    # Set Frontend ENV Variables Here
      - API_URL=http://mealie-api:9000
    restart: unless-stopped
    ports:
      - "9925:3000"
    volumes:
      - ./data:/app/data/
  mealie-api:
    image: hkotel/mealie:api-v1.0.0beta-5
    container_name: mealie-api
    mem_limit: 1000m
    volumes:
      - ./data:/app/data/
    environment:
    # Set Backend ENV Variables Here
      - ALLOW_SIGNUP=true
      - PUID=1000
      - PGID=1000
      - TZ=Australia/Melbourne
      - MAX_WORKERS=1
      - WEB_CONCURRENCY=1
      - BASE_URL=https://mealie.domain.tld
    restart: unless-stopped
```

After:
```
version: "2"
services:
  mealie:
    image: ghcr.io/mealie-recipes/mealie:nightly
    container_name: mealie
    ports:
      - "9925:9000"
    mem_limit: 1000m
    volumes:
      - ./data:/app/data/
    environment:
    # Set Backend ENV Variables Here
      - ALLOW_SIGNUP=true
      - PUID=1000
      - PGID=1000
      - TZ=Australia/Melbourne
      - MAX_WORKERS=1
      - WEB_CONCURRENCY=1
      - BASE_URL=https://mealie.domain.tld
    restart: unless-stopped
```

Image diff might be easier to take in:
![image](https://github.com/mealie-recipes/mealie/assets/3479092/085ebc5a-e245-4004-80dd-80aea2b8e647)

**Comment by djmega152** [comment]
My beta version was on sqlite and I'd like to migrate to postgres.  Is
there a way to do that?

**Comment by michael-genson** [comment]
Backups are db agnostic, so you can restore from backup using any db you want

**Comment by djmega152** [comment]
I deployed a single container using this guide
<https://mariushosting.com/how-to-install-mealie-v1-on-your-synology-nas/> and
as part of that I left my existing docker containers for the beta
deployment and created a mealiev1 folder.

The service starts properly and I am able to login.  When I go to Settings
> Backups, I tried to upload the backup that I did from the beta version
and nothing happens.

Here is the log
[image: image.png]
Next I uploaded the beta backup to the data/backups directory and then
refreshed the backup page.  I clicked and got the warning.  I checked the
checkbox and then clicked restore.  Here is the error I get. I assume this
is because I didn't run the SQL to update the user.  I don't know how to
get to a client to access the DB within docker/container manager.  Any
documentation that you have created that walks through that?

[image: image.png]

**Comment by djmega152** [comment]
Follow-up on this.  I finally figured out how to run the alter statement
but I am still getting the same 500 internal server error.

I'd love to get to the most recent release rather than staying on the beta
but I really don't want to rebuild all of my recipes manually.  I've tried
to follow the documentation and I know I've asked a lot of questions.
Please know I really appreciate the help and hope that we can find a
solution.  If there is anything you need me to provide, I'll happily do so.

Thanks!
Tony

**Comment by boc-the-git** [comment]
@djmega152 I think what I've taken in from all the stuff you've said is that you currently/previously have been running the V1beta5 (B5)? And are wanting to move to V1RC1 (RC1)?
You've also spoken about moving to postgres from SQLite.

I'd suggest you delay any change to your database, it's a complication you don't need right now.

I can't speak to any Synology intricacies as I've never used it, but by the sound of it you've already got a data folder mounted into your B5 instance? And you also spoke about creating a new folder for your RC1 instance. Migration could be as simple as copying that data folder from your B5 folder to your RC1 folder.
Then my prior message shows a basic change of a B5 compose file to a nightly variant (which is near identical to RC1). The guide you're trying to follow might also be sufficient for that RC1 compose file.

Note: I'm not seeing any images you might be trying to attach. I gather you might be responding via email, which might be the cause.

**Comment by djmega152** [comment]
Thank you for getting back to me and sorry for the delay responding.  I'll take your advice and hold off on converting to postgres for now.  

Here is a little background.  I installed via docker using [these](https://mariushosting.com/how-to-install-mealie-on-your-synology-nas/) instructions.  I don't have  adata folder like in the new deployment but instead everything is in the default /mealie directory.  
![image](https://github.com/mealie-recipes/mealie/assets/99622018/3b20094e-8229-4f3f-b3d0-f96d65ea41c6)

To ensure that I don't destroy my deployment, how would you suggest I proceed?

**Comment by djmega152** [comment]
Ok I got it!  

So last question (hopefully)....now that I am up and running on the latest release, is there a way to convert from SQLite to Postgres?

**Comment by djmega152** [comment]
I should have been more clear, I was able to update the version of the container to the latest.  The original bug I reported (and confirmed in the demo deployment) persists even on the updated version.

Here is the recipe setting I am testing with:
![image](https://github.com/mealie-recipes/mealie/assets/99622018/9328bff1-e27f-4364-8c5a-1674fc24f7f9)

Here is the user profle:
![image](https://github.com/mealie-recipes/mealie/assets/99622018/7f435ecc-7664-449d-87b8-55da8014377f)

The group the user is a part of:
![image](https://github.com/mealie-recipes/mealie/assets/99622018/4da51476-26d0-4082-a9c6-dc85236a6f6c)

And the result for that user.  This shows the search but the main page is blank for recipes as well:
![image](https://github.com/mealie-recipes/mealie/assets/99622018/6a21afd9-3fec-427c-85f1-335e729a5b93)

**Comment by djmega152** [comment]
Any update on this?

**Comment by michael-genson** [comment]
Have you added any recipes to the "Work Friends" group? If not, you won't see any recipes there. Recipes are not shared between groups (see my note above)

**Comment by djmega152** [comment]
How do I add a recipe to a group?  I've looked through the docs and I've edited existing recipes to see if I can assign them but I don't see any Group assignment option.  Am I missing something obvious?

**Comment by hay-kot** [comment]
> How do I add a recipe to a group? I've looked through the docs and I've edited existing recipes to see if I can assign them but I don't see any Group assignment option. Am I missing something obvious?

Recipes are created in whatever group the user that created them is in. There is not a way to share recipes across groups or change the group of a recipe.

**Comment by djmega152** [comment]
Thank you for that info hay-kot.  I'll log an enhancement request as I think it would be really useful to be able to share especially if you have multiple groups.

**Comment by Kuchenpirat** [comment]
there is already a discussion that should pretty much cover your request. 
Take a look at #2134

**Comment by hay-kot** [comment]
Pretty sure the issue is resolved and the feature work in #2134 covers the rest of this request. Feel free to re-open if there's more to resolve here.
